### DISCUSSION TITLE
Some of the records are not getting inserted in the table when I set pool_pre_ping=True in create_engine()

### DISCUSSION THREAD

**Discussion by srkhedkar** [sqlalchemy/sqlalchemy#10511] [Usage Questions]
Hi,

To fix the psycopg2.OperationalError: server closed the connection unexpectedly, I started passing pool_pre_ping=True argument while creating the engine. However , after making this change I observed that while inserting records in the table , only some records get inserted and others get skipped. I do not get any SQL related errors. When , I remove " pool_pre_ping=True" everything works fine. 

I am using
SQLAlchemy==1.4.31
psycopg2-binary==2.9.5
fastapi==0.101.0
PostgreSQL DB

Is this a Bug or am I missing something ?

**Comment by zzzeek** [comment]
hi -

There's no way for me to know if this is a bug or not without at reproducing test case.    ping would normally have nothing to do with transaction commits, however, the ping routine for the psycopg drivers necessarily has to manipulate the autocommit element of the connection.  But this is restored when the routine completes - and even then, if the connection were somehow inadvertently in autocommit mode, that wouldn't cause records to not be committed.

typically when users are having problems with fastAPI it's due to incorrect use of the`Session` where it gets shared between requests, leading to concurrent methods invoked on a single connection with unpredictable results.     this is why it's important to isolate the behavior.
